### PR TITLE
Migrate mp4 decoding demo

### PR DESCRIPTION
**Pull request by tguilbert-google**
This PR migrates the mp4 extraction demo from:
https://tguilbert-google.github.io/webcodecs/mp4/index.html

### CODE REVIEW

File: samples/index.html [hunk 1]
```diff
@@ -11,9 +11,9 @@
...
   <div id="wrapper">
     <div id="content">
-    <article onclick="window.location.href='/';">
-      <h1><a href="/">Sample Demo 1</a></h1>
-      <p>A sample that shows cool WebCodecs stuff.</p>
+    <article onclick="window.location.href='/samples/mp4-decode/';">
+      <h1><a href="/samples/mp4-decode/">MP4 Decoding</a></h1>
+      <p>Demuxes (using mp4box.js) and decodes an mp4 file.</p>
     </article>
     <article onclick="window.location.href='/';">
```
Comment (dalecurtis) [inline samples/index.html:18]: I'd just delete the rest of these place holder ones now that there's a real entry :)

File: samples/mp4-decode/index.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+<body>
+  <p>
+    This demo extracts all frames from an MP4 file and renders them to a canvas as fast as possible. It uses <a href="https://github.com/gpac/mp4box.js/">mp4box.js</a> to parse and demux the file.
+  </p>
+  <p>
+    Frames extracted: <span id="frameCount"></span>
+  </p>
+  <canvas></canvas>
+
+<script src="mp4box.all.min.js"></script>
```
Comment (dalecurtis) [inline samples/mp4-decode/index.html:17]: Move script out of body or otherwise fix indenting?

File: samples/index.html (hunk 1 above)
Comment (dalecurtis) [reply samples/index.html:18]: Actually since we're all working on this never mind, easier to merge multiple commits :)

Comment (tguilbert-google) [reply samples/index.html:18]: Seems good.

Comment (tguilbert-google) [reply samples/index.html]: Hrm. Deleting a commit with Windows Github Desktop is not as easy as I hoped...

**Comment by tguilbert-google** [comment]
Abandoning this PR in favor of #219

File: samples/mp4-decode/index.html (hunk 2 above)
Comment (tguilbert-google) [reply samples/mp4-decode/index.html:18]: Done (in #219)

File: samples/index.html (hunk 1 above)
Comment (tguilbert-google) [reply samples/index.html]: I created a new branch. I'm still haven't fixed my main though.

**Pull request closed without merging**